### PR TITLE
fix(kit): InputPhoneInternational fix drop/paste handler

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/input-phone-international/test/international-input-phone.component.spec.ts [hunk 1]
```diff
@@ -143,6 +143,30 @@ describe('InputPhoneInternational', () => {
...
+
+            component.onPaste(pasteEvent);
+
+            expect(component.countryIsoCode).toBe(TuiCountryIsoCode.RU);
+        });
+
+        it('should replace code on paste event ', () => {
+            const data = new DataTransfer();
+
+            data.setData('text/plain', '88005553535');
```
Comment (splincode) [inline projects/kit/components/input-phone-international/test/international-input-phone.component.spec.ts:161]: can we add a test where the situation might be: '+89112223344'

Comment (vladimirpotekhin) [reply projects/kit/components/input-phone-international/test/international-input-phone.component.spec.ts:161]: done 👍

File: projects/kit/components/input-phone-international/input-phone-international.component.ts [hunk 2]
```diff
@@ -209,4 +213,18 @@ export class TuiInputPhoneInternationalComponent
         this.countryIsoCode = code;
         this.countryIsoCodeChange.emit(code);
     }
+
+    private extractCountryCode(value: string): TuiCountryIsoCode | undefined {
+        return this.countries.find(countryIsoCode => {
+            const ruCodeTest =
```
Comment (nsbarsukov) [inline projects/kit/components/input-phone-international/input-phone-international.component.ts:219]: we have more cases.
For example, Japan starts with `81-##-####-####`

Let takes any `8 (812) xxx-xx-xx` (Saint-Petersburg).
Paste will not work too(

Comment (nsbarsukov) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts:219]: oh, stop, sorry.

I have made checks on `taiga-ui.dev`, not feature stand

Comment (splincode) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts:219]: you can add a test for this case

Comment (nsbarsukov) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts:219]: yeah, i agree that adding some tests with phone numbers similar to these countries will be cool

```
    [TuiCountryIsoCode.CN]: {mask: '+86(###) ####-####', name: 'Китайская Н.Р.'},
    [TuiCountryIsoCode.HK]: {mask: '+852####-####', name: 'Гонконг'},
    [TuiCountryIsoCode.JP]: {mask: '+81-##-####-####', name: 'Япония '},
    [TuiCountryIsoCode.KH]: {mask: '+855##-###-###', name: 'Камбоджа'},
    [TuiCountryIsoCode.KP]: {mask: '+850####-#############', name: 'Корейская НДР'},
    [TuiCountryIsoCode.KR]: {mask: '+82##-###-####', name: 'Респ. Корея'},
    [TuiCountryIsoCode.LA]: {mask: '+856##-##-###-###', name: 'Лаос'},
    [TuiCountryIsoCode.MO]: {mask: '+853####-####', name: 'Макао'},
    [TuiCountryIsoCode.TW]: {mask: '+886#-####-####', name: 'Тайвань'},
    [TuiCountryIsoCode.VN]: {mask: '+84(###) ####-###', name: 'Вьетнам'},
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/input-phone-international/input-phone-international.component.ts (hunk 2 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts:219]: added for Taiwan and Rostov

**Review by waterplea — APPROVED** [review]

**Pull request merged**